### PR TITLE
fix(k8s): make WaitForPool(s) coherent with others

### PR DESCRIPTION
**Pull request by Sh4d1**
Internal ticket APIGW-1949

Signed-off-by: Patrik Cyvoct <pcyvoct@scaleway.com>

### CODE REVIEW

File: api/k8s/v1beta4/k8s_helpers.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ func (s *API) WaitForClusterPools(req *WaitForClusterPoolsRequest) error {
 	}
 
 	for _, pool := range pools.Pools {
-		err = s.WaitForPool(&WaitForPoolRequest{
+		_, err := s.WaitForPool(&WaitForPoolRequest{
```
Comment (kindermoumoute) [inline api/k8s/v1beta4/k8s_helpers.go]: pool object should be used, and `WaitForClusterPools ` should return []*Poll

**Comment by kindermoumoute** [comment]
need rebase

File: api/k8s/v1beta4/k8s_helpers.go (hunk 1 above)
Comment (Sh4d1) [reply api/k8s/v1beta4/k8s_helpers.go]: Hmm cf https://github.com/scaleway/scaleway-sdk-go/pull/334 ?

File: api/k8s/v1beta4/k8s_helpers.go [hunk 2]
```diff
@@ -57,13 +57,6 @@ func (s *API) WaitForCluster(req *WaitForClusterRequest) (*Cluster, error) {
...
 
-// WaitForClusterPoolsRequest is used by WaitForClusterPools method.
-type WaitForClusterPoolsRequest struct {
-	ClusterID string
-	Region    scw.Region
-	Timeout   *time.Duration
-}
-
 // WaitForPoolRequest is used by WaitForPool method.
 type WaitForPoolRequest struct {
```
Comment (kindermoumoute) [inline api/k8s/v1beta4/k8s_helpers.go]: delete this

**Pull request merged**